subiquity/server/controllers/filesystem.py

```diff
@@ -461,19 +460,36 @@ async def _probe(self, *, context=None):
                 continue
             break
 
+    def run_guided(self, layout):
+        guided_method = getattr(self, "guided_" + layout['name'])
+        mode = layout.get('mode', 'reformat_disk')
+        self.validate_layout_mode(mode)
+
+        if mode == 'reformat_disk':
+            match = layout.get("match", {'size': 'largest'})
+            disk = self.model.disk_for_match(self.model.all_disks(), match)
+            if not disk:
+                raise Exception("autoinstall cannot configure storage "
+                                "- no disk found large enough for install")
+        elif mode == 'use_gap':
+            gap = self.largest_gap(self.model.all_disks())
```

Comment:
This should filter the disks through "can_be_boot(d, with_reformatting=False)" I think. In general the device selection here should probably follow FilesystemController.guided_GET more closely but well.  autoinstall into the biggest gap on a VROC subvolume isn't something that has come up as a feature request yet.